src/realm/parser/expression_container.cpp

```diff
@@ -209,6 +217,9 @@ DataType ExpressionContainer::check_type_compatibility(DataType other_type)
                         data_type_to_str(other_type)));
     }
     else if (self_type.value() != other_type) {
+        if (types_are_numerically_comparible(self_type.value(), other_type)) {
+            return other_type; // FIXME: type cast to most precise
```

Comment:
Hrm good catch. I think I will just remove this because comparing a column to a column of different numeric type actually doesn't work right now, even though I started adding support here but didn't finish. As you point out, it will take some careful though on how we do the type casts :(